src/util/clickTracking.js

```diff
@@ -0,0 +1,154 @@
+const DATA_ATTR = 'clicktrack';
+
+function cleanTrackingUrl() {
+	// TODO: investigate and refactor to support this feature
+
+	/*
+	// remove tracking information from the url to prevent bad link copying
+	// and to clean up the url field
+
+	var TRACKING_PARAM = '_xtd';
+	if ( history && history.replaceState && window.location.href.indexOf(TRACKING_PARAM) > 0) {
+
+		var baseUrl = window.location.href.split('?');
+
+		//Make sure there was actually a query param.
+		if (!baseUrl[1]) {
+			return;
+		}
+
+		var qs = baseUrl[1].split('&');
+
+		for (var i = 0, len = qs.length; i < len; i++) {
+			if (qs[i] && qs[i].split('=')[0] === TRACKING_PARAM) {
+				qs.splice(i, 1);
+				break;
+			}
+		}
+
+		history.replaceState({}, document.title, baseUrl[0] + '?' + qs.join('&'));
+	}
+	*/
+}
+
+// set reference to un-modified stopPropagation
+const originalStopPropagation = typeof Event !== 'undefined' && Event.prototype.stopPropagation;
+
+/**
+ * Overriding stopPropagation for maximum click tracking
+ * @param {Function} trackClick function to track clicks
+ * @return {undefined} side effects only
+ */
+export function trackStopPropagation(trackClick) {
+	/**
+	 * A special jQuery events method that hooks the click tracking into events
+	 * that would never bubble to document.body
+	 */
+	Event.prototype.stopPropAndTrack = function() {
+		trackClick(this);
+		return originalStopPropagation.call(this);
+	};
+
+	// override stopPropagation to track clicks
+	Event.prototype.stopPropagation = function() {
+		if (this.type === 'click') {
+			trackClick(this);
+		}
+		return originalStopPropagation.call(this);
+	};
+}
+
+/**
+ * Get a descriptive string for the HTML element passed in, e.g.
+ *
+ * `tagName#id.class1.class2`
+ *
+ * @param {HTMLElement} element the element to describe
+ * @param {Boolean} idOnly only use id in description
+ */
+function elementShorthand(element, idOnly) {
+	const id = element.id ? `#${element.id}` : '';
+	if (id && idOnly) {
```

Comment:
Since the classnames might be updated dynamically in response to user activity, there are probably cases where you would like to know if it's the `#magicButton.active` instead of the `#magicButton.disabled`, for example